src/main/scala/org/broadinstitute/dsde/firecloud/service/DataUseRestrictionSupport.scala

```diff
@@ -58,6 +61,86 @@ trait DataUseRestrictionSupport extends LazyLogging {
+    val nmds = duosDataUse.methodsResearch.map ( AttributeName.withLibraryNS("NMDS") -> AttributeBoolean(_) )
+
+    // FC supports both NCU and NPU; DUOS only supports one (NCU)
+    // val npu = duosDataUse.commercialUse.map ( AttributeName.withLibraryNS("NPU") -> AttributeBoolean(_) )
+
+    // DUOS represents NAGR and NCTRL as strings ("yes"|"no")
+    val nagr = duosDataUse.aggregateResearch match {
+      case Some(s) if "yes".equals(s.toLowerCase) => Some(AttributeName.withLibraryNS("NAGR") -> AttributeBoolean(true))
+      case Some(s) if "no".equals(s.toLowerCase) => Some(AttributeName.withLibraryNS("NAGR") -> AttributeBoolean(false))
+      case _ => None
+    }
+    val nctrl = duosDataUse.controlSetOption match {
+      case Some(s) if "yes".equals(s.toLowerCase) => Some(AttributeName.withLibraryNS("NCTRL") -> AttributeBoolean(true))
+      case Some(s) if "no".equals(s.toLowerCase) => Some(AttributeName.withLibraryNS("NCTRL") -> AttributeBoolean(false))
+      case _ => None
+    }
+
+    // DUOS has no concept of "IRB review required"
+    // val irb = None
+
+    // disease restrictions: DS, DS_URL
+    val ds = duosDataUse.diseaseRestrictions match {
+      case Some(Seq()) => Map.empty[AttributeName,Attribute]
+      case Some(nodeList) =>
+        Map(
+          AttributeName.withLibraryNS("DS_URL") -> AttributeValueList(nodeList.map( AttributeString )),
+          AttributeName.withLibraryNS("DS") -> AttributeValueList(nodeList.map{ nodeid =>
+            ontologyDAO.search(nodeid) match {
+              case termResource :: Nil => AttributeString(termResource.label)
+              case _ => AttributeString(nodeid)
+            }})
+        )
+      case _ => Map.empty[AttributeName,Attribute]
+    }
+
+    // population restrictions: RS-POP
+    val rspop = duosDataUse.populationRestrictions match {
+      case Some(Seq()) => Map.empty[AttributeName,Attribute]
+      case Some(populations) =>
+        Map(
+          AttributeName.withLibraryNS("RS-PD") -> AttributeValueList(populations.map(AttributeString))
+        )
+      case _ => Map.empty[AttributeName,Attribute]
+    }
+
+    // gender restrictions: RS-G, RS-FM, RS-M
+    val rsg = duosDataUse.gender match {
+      case Some(f:String) if "female".equals(f.toLowerCase) => Map(
+        AttributeName.withLibraryNS("RS-G") -> AttributeBoolean(true),
+        AttributeName.withLibraryNS("RS-FM") -> AttributeBoolean(true)
+      )
+      case Some(m:String) if "male".equals(m.toLowerCase) => Map(
+        AttributeName.withLibraryNS("RS-G") -> AttributeBoolean(true),
+        AttributeName.withLibraryNS("RS-M") -> AttributeBoolean(true)
+      )
+      case _ => Map.empty[AttributeName,Attribute]
+    }
+
+    val result = Map.empty[AttributeName, Attribute] ++ gru ++ hmb ++ ncu ++ nmds ++
+      rspd ++ nagr ++ nctrl ++ ds ++ rspop ++ rsg // ++ npu ++ irb
+
+    logger.debug("inbound DuosDataUse: " + duosDataUse)
+    logger.debug("outbound attrs: " + result)
```

Comment:
Do you still want these (here and elsewhere)? I'm fine if we keep them around for some period of time (they will be super helpful for debugging), but maybe a TODO to eliminate in X months?